### ISSUE TITLE
[Bug]: The index state is always IndexState_IndexStateNone when creating index asynchronously

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus-sdk-go#361]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

go sdk version: master
milvus image: 2.2.0-20221123-2966710b

Case
```
func TestCreateIndexAsync(t *testing.T) {
	ctx, cancel := context.WithTimeout(context.Background(), time.Second*common.DefaultTimeout*4)
	defer cancel()
	// connect
	mc := createMilvusClient(ctx, t)

	// create default collection with flush data
	collName, _ := createCollectionWithDataIndex(ctx, t, mc, false, false)

	// create index
	idx, _ := entity.NewIndexHNSW(entity.L2, 8, 96)
	mc.CreateIndex(ctx, collName, common.DefaultFloatVecFieldName, true, idx)

	cost, _ := time.ParseDuration("60s")
	start := time.Now()
	for true {
		time.Sleep(time.Second * 10)
		indexState, err := mc.GetIndexState(ctx, collName, common.DefaultFloatVecFieldName)
		common.CheckErr(t, err, true)
		if indexState == 3 {
			break
		}
		if time.Now().After(start.Add(cost)) {
			t.Fatal("Create index cost more than 60s")
		}
	}
}
```

case result:
```
2022/11/23 18:34:50 milvus_client.go:15: (ApiRequest): func [CreateIndex], args: [context.Background.WithDeadline(2022-11-23 18:38:45.952178888 +0800 CST m=+240.003286900 [3m55.567504399s]) PwrhnW float_vec true 0xc000424640 []]
2022/11/23 18:34:50 milvus_client.go:22: (ApiResponse): func [CreateIndex], results: []
2022/11/23 18:35:00 milvus_client.go:15: (ApiRequest): func [GetIndexState], args: [context.Background.WithDeadline(2022-11-23 18:38:45.952178888 +0800 CST m=+240.003286900 [3m45.508671198s]) PwrhnW float_vec []]
2022/11/23 18:35:00 milvus_client.go:22: (ApiResponse): func [GetIndexState], results: [0]
2022/11/23 18:35:10 milvus_client.go:15: (ApiRequest): func [GetIndexState], args: [context.Background.WithDeadline(2022-11-23 18:38:45.952178888 +0800 CST m=+240.003286900 [3m35.478140716s]) PwrhnW float_vec []]
2022/11/23 18:35:10 milvus_client.go:22: (ApiResponse): func [GetIndexState], results: [0]
2022/11/23 18:35:20 milvus_client.go:15: (ApiRequest): func [GetIndexState], args: [context.Background.WithDeadline(2022-11-23 18:38:45.952178888 +0800 CST m=+240.003286900 [3m25.448175499s]) PwrhnW float_vec []]
2022/11/23 18:35:20 milvus_client.go:22: (ApiResponse): func [GetIndexState], results: [0]
2022/11/23 18:35:30 milvus_client.go:15: (ApiRequest): func [GetIndexState], args: [context.Background.WithDeadline(2022-11-23 18:38:45.952178888 +0800 CST m=+240.003286900 [3m15.421829518s]) PwrhnW float_vec []]
2022/11/23 18:35:30 milvus_client.go:22: (ApiResponse): func [GetIndexState], results: [0]
2022/11/23 18:35:40 milvus_client.go:15: (ApiRequest): func [GetIndexState], args: [context.Background.WithDeadline(2022-11-23 18:38:45.952178888 +0800 CST m=+240.003286900 [3m5.393039416s]) PwrhnW float_vec []]
2022/11/23 18:35:40 milvus_client.go:22: (ApiResponse): func [GetIndexState], results: [0]
2022/11/23 18:35:50 milvus_client.go:15: (ApiRequest): func [GetIndexState], args: [context.Background.WithDeadline(2022-11-23 18:38:45.952178888 +0800 CST m=+240.003286900 [2m55.36441179s]) PwrhnW float_vec []]
2022/11/23 18:35:50 milvus_client.go:22: (ApiResponse): func [GetIndexState], results: [0]
    index_test.go:275: Create index cost more than 60s
2022/11/23 18:35:50 milvus_client.go:15: (ApiRequest): func [DropCollection], args: [context.Background.WithDeadline(2022-11-23 18:38:45.952178888 +0800 CST m=+240.003286900 [2m55.335183703s]) PwrhnW]
2022/11/23 18:35:50 milvus_client.go:20: (ApiResponse): func [DropCollection], error: rpc error: code = Canceled desc = context canceled
```

**Comment by ThreadDao** [comment]
Even if create index with `async=false`, the index state returned is also 0
```
func TestIndexState(t *testing.T) {
	ctx, cancel := context.WithTimeout(context.Background(), time.Second*common.DefaultTimeout)
	defer cancel()
	// connect
	mc := createMilvusClient(ctx, t)

	// create default collection with flush data
	collName, _ := createCollectionWithDataIndex(ctx, t, mc, false, false)

	// create index
	idx, _ := entity.NewIndexHNSW(entity.L2, 8, 96)
	err := mc.CreateIndex(ctx, collName, common.DefaultFloatVecFieldName, false, idx)
	common.CheckErr(t, err, true)

	// get index state
	state, errState := mc.GetIndexState(ctx, collName, common.DefaultFloatVecFieldName)
	common.CheckErr(t, errState, true)
	require.Equal(t, entity.IndexState(common.IndexStateValue["Finished"]), state, "Expected finished index state")
}
```
Result:
```
2022/11/23 18:49:17 milvus_client.go:15: (ApiRequest): func [CreateIndex], args: [context.Background.WithDeadline(2022-11-23 18:50:13.363707064 +0800 CST m=+60.002427685 [56.137733786s]) GpsMWe float_vec false 0xc000562c80 []]
2022/11/23 18:49:18 milvus_client.go:22: (ApiResponse): func [CreateIndex], results: []
2022/11/23 18:49:18 milvus_client.go:15: (ApiRequest): func [GetIndexState], args: [context.Background.WithDeadline(2022-11-23 18:50:13.363707064 +0800 CST m=+60.002427685 [54.621078001s]) GpsMWe float_vec []]
2022/11/23 18:49:18 milvus_client.go:22: (ApiResponse): func [GetIndexState], results: [0]
    index_test.go:303: 
        	Error Trace:	/home/zong/zong/go-projects/milvus-sdk-go/test/testcases/index_test.go:303
        	Error:      	Not equal: 
        	            	expected: 3
        	            	actual  : 0
        	Test:       	TestIndexState
```

**Comment by ThreadDao** [comment]
Fixed `master-20230816-8b549142`
